plugin/keyring_vault/vault_keyring.cc

```diff
@@ -47,7 +74,13 @@ int check_keyring_file_data(MYSQL_THD thd  MY_ATTRIBUTE((unused)),
 
   try
   {
-    boost::movelib::unique_ptr<IVault_curl> vault_curl(new Vault_curl(logger.get()));
+    if (reset_curl())
+    {
+      logger->log(MY_ERROR_LEVEL, "Cannot set keyring_vault_config_file");
+      mysql_rwlock_unlock(&LOCK_keyring);
```

Comment:
at least 5 places only in this function with `mysql_rwlock_unlock(&LOCK_keyring);` and `return xxx`.
Please, add a RAII capsule for locking / unlocking.